src/Windows/Avalonia.Win32/Win32Platform.cs

```diff
@@ -109,7 +109,7 @@ public static void Initialize(Win32PlatformOptions options, IDispatcherImpl cust
                 .Bind<IKeyboardDevice>().ToConstant(WindowsKeyboardDevice.Instance)
                 .Bind<IPlatformSettings>().ToConstant(s_instance)
                 .Bind<IPlatformThreadingInterface>().ToConstant(s_instance)
-                .Bind<IDispatcherImpl>().ToConstant(customDispatcher ?? DispatcherImpl.UIThread)
+                .Bind<IDispatcherImpl>().ToConstant(customDispatcher ?? new DispatcherImpl(AvaloniaLocator.Current.GetService<IPlatformThreadingInterface>()))
```

Comment:
can we just use `s_instance` instead of `AvaloniaLocator.Current.GetService<IPlatformThreadingInterface>()`?

Reply:
yes, my bad, we can. Here is the update https://github.com/JetBrains/Avalonia/pull/4